test/e2e/data/user.go

```diff
@@ -0,0 +1,79 @@
+package data
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+)
+
+const (
+	// build-in dbroles
+	RoleBuildInAdmin        string = "atlasAdmin"
+	RoleBuildInReadWriteAny string = "readWriteAnyDatabase"
+	RoleBuildInReadAny      string = "readAnyDatabase"
+
+	DefaultDatabaseName = "admin"
+)
+
+func BasicUser(crName, atlasUserName string, add ...func(user *v1.AtlasDatabaseUser)) *v1.AtlasDatabaseUser {
+	user := &v1.AtlasDatabaseUser{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crName,
+		},
+		Spec: v1.AtlasDatabaseUserSpec{
+			Project: common.ResourceRefNamespaced{
+				Name: ProjectName,
+			},
+			Username: atlasUserName,
+		},
+	}
+	for _, f := range add {
+		f(user)
+	}
+	return user
+}
+
+func WithSecretRef(name string) func(user *v1.AtlasDatabaseUser) {
+	return func(user *v1.AtlasDatabaseUser) {
+		user.Spec.PasswordSecret = &common.ResourceRef{Name: name}
+	}
```

Comment:
Why to return a function instead of an AtlasDatabaseUser object?

Reply:
This is a function passed to the user struct constructor. So that you can customize the standard user at the time the constructor is called.